boot/bootstate20.go

```diff
@@ -195,6 +196,14 @@ type bootState20Kernel struct {
 	dev Device
 }
 
+func (ks20 *bootState20Kernel) getRebootBootloader() bootloader.RebootBootloader {
+	if err := ks20.loadBootenv(); err != nil {
+		// log and keep going while returning nil
+		logger.Noticef("cannot load boot environement: %v", err)
```

Comment:
what situation do we anticipate this to be the case? why not just return an error here (and adjust the `getRebootBootloader` method on the interface to also return an error?

also small typo:

```suggestion
		// log and keep going while returning nil
		logger.Noticef("cannot load boot environment: %v", err)
```

Reply:
`loadBootenv()` can error out if no bootloader is found, or if it cannot load the boot env vars. I always doubt with this sort of situations because on one hand you could still be able to reboot even if you don't know if you have a `bootloader.RebootBootloader`, but on the other hand if an error happens it implies that the system is badly broken.

Anyway, thinking about it, maybe in this case it makes more sense to return the error, so link-snap fails and the new kernel is not installed, which is probably good if the system is broken. So I'll make it return an error.

Reply:
Thanks, yeah I would much prefer we be very careful with ignoring/not propagating errors since the boot package is very critical and for the most part if an error is encountered something will likely be undone which hopefully can restore the system at least partially